lib/mayaUsd/fileio/primUpdaterManager.cpp

```diff
@@ -284,61 +285,121 @@ PullImportPaths pullImport(
     MayaUsd::ImportData importData(mFileName);
     importData.setRootPrimPath(pulledPrim.GetPath().GetText());
 
-    UsdMaya_ReadJob readJob(importData, jobArgs);
-    auto            found = userArgs.find("Maya:Pull:ParentPath");
-    if (found != userArgs.end()) {
-        const std::string& dagPathStr = found->second.Get<std::string>();
-        auto               pullParent = UsdMayaUtil::nameToDagPath(dagPathStr);
-        if (pullParent.isValid()) {
-            readJob.SetMayaRootDagPath(pullParent);
+    auto readJob = std::make_shared<UsdMaya_ReadJob>(importData, jobArgs);
+
+    MDagPath pullParentPath;
+    {
+        auto found = userArgs.find(kPullParentPathKey);
+        if (found != userArgs.end()) {
+            const std::string& dagPathStr = found->second.Get<std::string>();
+            pullParentPath = UsdMayaUtil::nameToDagPath(dagPathStr);
+            if (pullParentPath.isValid()) {
+                readJob->SetMayaRootDagPath(pullParentPath);
+            }
         }
     }
 
     // Execute the command, which can succeed but import nothing.
-    bool success = readJob.Read(&addedDagPaths);
+    bool success = readJob->Read(&addedDagPaths);
     if (!success || addedDagPaths.size() == 0) {
         return PullImportPaths({}, {});
     }
 
+    // Note: UsdMaya_ReadJob has explicit Read(), Undo() and Redo() functions,
+    //       and Read() has already been called, so create the function-undo item
+    //       but do not execute it.
+    FunctionUndoItem::create(
```

Comment:
Most of the changes in the PrimUpdaterManager are to convert direct C++ calls to OpUndoItem creation. Some previously non-undoable stuff had to be converted to uodobale and some operations had to be refined or reordered to work with undo.